DOCSP-49224 Go v2.2 release (#486)
mongodb/docs-golang: source/whats-new.txt

  see the example in the :ref:`BSON Options <golang-bson-options>` section of the 
  BSON guide. 

- Adds support for Kubernetes OIDC authentication. To learn more about how to configure 
  MongoDB authentication for a Kubernetes cluster, see the Kubernetes section of the 

>   :ref:`golang-enterprise-authentication-mechanisms` guide.

Reviewer: S: consider adding an anchor on the specific section and linking there instead

- Adds support to configure the lifetime of the Data Encryption Key (DEK) cache. To learn
  more about DEK management, see the 
  :manual:`Encryption Key Management </csfle/fundamentals/manage-keys>` guide in the 
  {+mdb-server+} manual. 

> - Adds support for ``error.Is`` and ``error.As`` in all stable public APIs for improved 

Reviewer: S: consider removing as users usually only see the public APIs
```suggestion
- Adds support for the ``error.Is()`` and ``error.As()`` functions for improved 
```
